### PR TITLE
[WIP] feat(make): makefile dependency generation

### PR DESCRIPTION
**Pull request by rishabh3112**
**What kind of change does this PR introduce?**

feature #152 
**Did you add tests for your changes?**
No
**If relevant, did you update the documentation?**
Yes
**Summary**
N/A

**Does this PR introduce a breaking change?**

No
**Other information**
For my vision on this refer this [comment](https://github.com/webpack/webpack-cli/issues/152#issuecomment-435606753).

### CODE REVIEW

File: packages/info/tsconfig.json [hunk 1]
```diff
@@ -1,3 +1,3 @@
-{
-  "extends": "../../tsconfig.packages.json"
-}
+{
```
Comment (ematipico) [inline packages/info/tsconfig.json]: Again these changes... could please make sure that are not touched again? It's not the first time

File: packages/make/index.ts [hunk 2]
```diff
@@ -1,10 +1,34 @@
...
+ * @returns void
  *
  */
-
 export default function make() {
-	return;
+	const userDirectory: String = process.cwd();
+
+	let isBase: Boolean = false;
+	let dependencyTree: Object;
```
Comment (ematipico) [inline packages/make/index.ts]:
```suggestion
```

File: packages/make/index.ts [hunk 3]
```diff
@@ -1,10 +1,34 @@
...
+
+	if (fs.existsSync(__dirname + "/base")) {
+		isBase = true;
+	} else {
+		storeBase(userDirectory);
+	}
+
+	dependencyTree = getDependencyTree();
```
Comment (ematipico) [inline packages/make/index.ts]:
```suggestion
	const dependencyTree = getDependencyTree();
```

File: tsconfig.base.json [hunk 4]
```diff
@@ -1,11 +1,11 @@
...
-    "target": "es6",
-    "module": "commonjs",
-    "moduleResolution": "node",
-    "allowSyntheticDefaultImports": true,
-    "skipLibCheck": true
-  },
-  "include": ["packages/**/*.ts"],
-  "exclude": ["node_modules/**", "packages/*/node_modules/**", "packages/*/__tests__/*.test.ts"]
-}
+{
```
Comment (ematipico) [inline tsconfig.base.json]: Here too... let's remove them

File: packages/info/tsconfig.json (hunk 1 above)
Comment (rishabh3112) [reply packages/info/tsconfig.json]: I will try again. Can you guide me here?

Comment (ematipico) [reply packages/info/tsconfig.json]: We have a `.gitattributes` file. Could change it to `* text eol=lf` and try again?

**Comment by rishabh3112** [comment]
@evenstensberg can you find where the problem is in `storeBase()` function.

File: packages/make/utils/baseHandler.ts [hunk 5]
```diff
@@ -0,0 +1,59 @@
+import * as fs from "fs";
+import * as dir from "node-dir";
+/**
+ * @returns storeBase() , getDependenencyTree(), getChangedTree(), builder()
+ *
+ */
+
+const log = (text) => { process.stdout.write(text); };
```
Comment (evenstensberg) [inline packages/make/utils/baseHandler.ts]: actually I don't think you need brackets here

**Comment by evenstensberg** [comment]
Could you give a more elaborative description of your problem?

File: packages/make/utils/baseHandler.ts (hunk 5 above)
Comment (rishabh3112) [reply packages/make/utils/baseHandler.ts]: Having brackets is my habit, Removed now 👍

**Comment by rishabh3112** [comment]
Problem is now solved @evenstensberg.

**Comment by evenstensberg** [comment]
Awesome!

**Comment by rishabh3112** [comment]
Error at pipeline
```
baseHandler › storeBase › should store all js files
2018-11-28T15:35:37.8681389Z 
2018-11-28T15:35:37.8681565Z     TypeError: shell.test is not a function
2018-11-28T15:35:37.8681759Z 
2018-11-28T15:35:37.8681909Z        7 |         baseHandler_1.storeBase(process.cwd() + "../testProject");
2018-11-28T15:35:37.8682060Z        8 |         it("should store all js files", () => {
2018-11-28T15:35:37.8682558Z     >  9 |             expect(shell.test("-e", "../utils/base/index.js")).toBe(true);
2018-11-28T15:35:37.8682778Z          |                          ^
2018-11-28T15:35:37.8683185Z       10 |             expect(shell.test("-e", "../utils/base/webpack.config.js")).toBe(true);
2018-11-28T15:35:37.8683675Z       11 |             expect(shell.test("-e", "../utils/base/utils/foo.js")).toBe(true);
2018-11-28T15:35:37.8684102Z       12 |             expect(shell.test("-e", "../utils/base/utils/bar.js")).toBe(true);
2018-11-28T15:35:37.8684318Z 
2018-11-28T15:35:37.8684472Z       at Object.test (packages/make/__tests__/baseHandler.js:9:26)
2018-11-28T15:35:37.8684609Z 
```
but shelljs has a[ test function](https://github.com/shelljs/shelljs#testexpression).
Help!

**Comment by evenstensberg** [comment]
I'd say that you could dive into our codebase and do some console logs, play around in https://astexplorer.net/ to have a better understanding of how AST's work

**Comment by rishabh3112** [comment]
Thanks, @evenstensberg.
Had fun to explore ASTs!

**Comment by rishabh3112** [comment]
Hi @evenstensberg, Got a little busy on `UI` in between.
I wanted to know that how do I selectively build a file in webpack?

**Comment by rishabh3112** [comment]
Rest part done @evenstensberg.

**Comment by evenstensberg** [comment]
@sokra had some feedback for you regarding this

**Comment by rishabh3112** [comment]
Awesome @evenstensberg, @sokra I am looking forward to that!

**Comment by rishabh3112** [comment]
> Hi @evenstensberg, Got a little busy on `UI` in between.
> I wanted to know that how do I selectively build a file in webpack?

Hi, I was doing research on this. One of the solutions can be to make the `entry` for each and every file in the `webpack config` and when we have to build a segment of the dependency tree we can build a particular entry corresponding file.
Is there any other elegant solution existing for this specific task?

**Comment by evenstensberg** [comment]
you can focus on UI, this is not a #1 prio

**Comment by rishabh3112** [comment]
Currently, init- interface PR is struck and we don't have the designs so that we can start the frontend.
So, I can work on this for now 😃

**Comment by evenstensberg** [comment]
Alright, will try to get a design board up

**Pull request closed without merging**